[Phyo Han] iP

Nahoyhp opened the pull request.

Commit:
Add Gradle support

Nahoyhp committed:
add echo function

Nahoyhp committed:
create Task.java

Nahoyhp committed:
create TaskManager.java

Nahoyhp committed:
modify TaskManager.addTask

Arguement for addTask changes from Task to a String.

Nahoyhp committed:
modify TaskManager.toString

To get rid of the last empty line.

Nahoyhp committed:
incorporate TaskManager in Duke.java

Switch from if-else to switch.

Nahoyhp committed:
create Task.setDone()

Nahoyhp committed:
modify TaskManager to accomodate the Status of Task

Nahoyhp committed:
add compeleteTask features to Duke

Nahoyhp committed:
Change Task to abstract class

Nahoyhp committed:
create Deadline.java

Nahoyhp committed:
create ToDo.java

Nahoyhp committed:
create Event.java

Nahoyhp committed:
add different task creator to TaskManager.java

Nahoyhp committed:
create methods to prepare for different type of tasks

echoNewTask() : Print out Message for after adding a new Task
interpretInput(): Break down the input into 3 components [Command, Message, Extra_info]
    - Command: What the user intend to do
    - Message: Name of the Task
    - Extra_info: Extra info for the task. Example: Date to be done by or the time of the event

Nahoyhp committed:
create new cases for different types of tasks

Nahoyhp committed:
Modify error message for IndexOutOfBounds

Nahoyhp committed:
Change format for welcome message for easier regression testing

Nahoyhp committed:
modify interpretInput()

So that it now return varying size of 1,2 or 3 depending on the input.

Nahoyhp committed:
Fix malfunction in Duke.InterpretInput() when there is only 1 word

Nahoyhp committed:
create TaskDoneException

Nahoyhp committed:
set Task.setDone() to throw TaskDoneException when task is already done

Nahoyhp committed:
modify TaskManager to throw TaskDoneException

Nahoyhp committed:
Modify Duke to handle TaskDoneException

Nahoyhp committed:
Change formatting to display only ascii characters.

Prevent problem of .txt file not being display the check and the cross which reflect the status of the Task.

Nahoyhp committed:
Add label to ensure all commands are invalid after "bye"

Also make some modification to output styling

Nahoyhp committed:
Include new error message when user include / in the title of todo task

Nahoyhp committed:
add TaskManager.deleteTask()

Nahoyhp committed:
add new case, "del" , in switch

Nahoyhp committed:
add new message when no task is in the taskManager

Nahoyhp committed:
Add javadoc for methods in Duke.java

Nahoyhp committed:
Change indentation for Switch-Case.

Nahoyhp committed:
Correct indentation for switch cases.

Nahoyhp committed:
Create BadlyFormatted

Nahoyhp committed:
Create toSaveFormat() for Task and its sub-classes.

Nahoyhp committed:
Add FileRead and FileWrite for FileManager

Nahoyhp committed:
Add the ability to load and save to .txt file for Duke

Nahoyhp committed:
Add Date_FORMATTER & TIME_FORMATTER as static members for Task

Nahoyhp committed:
Change the data type for Time in Deadline from (String) to (Date, Time)

As a result, amendemnt is made to toString() as well.

Nahoyhp committed:
Split the Event's variable:  Detail from (String) to (LocalDate, StartTime, EndTime)

Change some formatting for ToString() method as well.

Nahoyhp committed:
Create new catch cases to adapt to the Exceptions occured when handling LocalDate & LocalTime

Nahoyhp committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Duke.java

Nahoyhp committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

Nahoyhp committed:
Create new formatter to format the Date properly in saved file.

As a result, changes the formatting to all Task's sub-classes toSaveFormat() methods

Nahoyhp committed:
Include Date as "Optional<LocalDate>" variables in Task.

Task:
- Added setDate() and getDate()

Task Sub-class:
- Changes the formatting for toString() and toSaveFormat()

Nahoyhp committed:
Rename TaskManager to TaskList

Nahoyhp committed:
Create DukeException to replace BadFormatFileException and TaskDoneException.

Nahoyhp committed:
Replace Exception thrown with DukeExceptions in Task and its sub-classes

Nahoyhp committed:
Modify TaskList to handle DukeException

Nahoyhp committed:
Create Storage and Ui class.

Nahoyhp committed:
Accomodate Storage, UI and DukeException into Duke.java

Nahoyhp committed:
Create Parser to handle user input

Nahoyhp committed:
Group classes into packages

Nahoyhp committed:
Move duke.txt from "src/data" to "data:

Nahoyhp committed:
Add TestCases for Task's sub-classes.

Nahoyhp committed:
Add ParserTest

Nahoyhp committed:
Add JUnit Test for Storage and TaskList

Nahoyhp committed:
Fix bugs in TaskList where last task can't be deleted

Nahoyhp committed:
Amend message for FileNotFound

Nahoyhp committed:
Stop tracking data/duke.txt

Nahoyhp committed:
Add JavaDoc for all the files.

Nahoyhp committed:
Move catch for all DukeExceptions to the bottom.

Nahoyhp committed:
Merge branch 'branch-A-CodingStandard'

Nahoyhp committed:
Add new function Search_By_KeyWord for Duke

Add:
- new switch-cases for Duke
- new find() for TaskList
- getName() for Task

Nahoyhp committed:
Merge branch 'branch-Level-9'

Resolve these conflicts:
#	src/main/java/Duke.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/task/Task.java

Nahoyhp committed:
Load file function create new directory when system can't find a file indicated.

Nahoyhp committed:
Changes method names to fit conventions.

Nahoyhp committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

Nahoyhp committed:
Rename the package names.

Nahoyhp committed:
Change Package location for Ui

BILLXYR commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,137 @@
+ */
+public class Duke {
+    private final Ui ui;
+    private TaskList taskList;
```
The textbook says: "Distinguish clearly between single-valued and multivalued variables."
Here I assume the taskList should be many tasks inside the list?
So perhaps it should be taskLists instead of taskList

Nahoyhp replied:
Yup. You are right. I have made amendment for all the suggestions that you made.

Thanks.

BILLXYR commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,83 @@
+    protected static final DateTimeFormatter SAVE_DATE_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    protected static final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ofPattern("HH:mm");
+
+    private final String name;
```
The book says: "A name is not just for differentiation; it should explain the named entity to the reader accurately and at a sufficient level of detail."
perhaps you can change the "name" variable to a more meaning name, such as taskName.

BILLXYR reviewed (commented):
Overall it is a very well-done code, just two minor problem, use plural form + more meaningful name. Good Job. 👍

Nahoyhp committed:
Create a basic GUI with no customization for Duke

dcchan98 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,116 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    private final Ui ui;
```
Should UI be named in caps as it does not change?

dcchan98 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,116 @@
+    public String processInput(String input) {
+        int index;
+        try {
+            String[] words = parser.splitIntoComponents(input);
```
Can this be a more meaningful identifier?

dcchan98 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,31 @@
+     * @return a String array that contains different components of the input.
+     */
+    public String[] splitIntoComponents(String input) {
+        ArrayList<String> list = new ArrayList<>();
```
You may want to indicate that it is list for inputs?

Nahoyhp replied:
Thanks for the feedback. I have made the amendment you suggested. Thanks

dcchan98 reviewed (commented):
Looks great to me. Nice job on the code!

Nahoyhp committed:
Repackage the files into duke and graphicInterface.

Nahoyhp committed:
Install Checkstyle with Gradle.

Amend the most of the violations by the process.

euzintan commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,31 @@
+     * @param input The input from the users.
+     * @return a String array that contains different components of the input.
+     */
+    public String[] splitIntoComponents(String input) {
+        ArrayList<String> inputComponents = new ArrayList<>();
+        int spaceIndex = input.indexOf(" ");
+        int slashIndex = input.indexOf("/");
+        int infoIndex = input.indexOf(" ", slashIndex);
+        if (spaceIndex == -1) {
+            inputComponents.add(input);
+        } else if (slashIndex == -1) {
+            inputComponents.add(input.substring(0, spaceIndex));
+            inputComponents.add(input.substring(spaceIndex + 1));
+        } else {
+            inputComponents.add(input.substring(0, spaceIndex));
+            inputComponents.add(input.substring(spaceIndex + 1, slashIndex));
+            inputComponents.add(input.substring(infoIndex + 1));
+        }
+        return inputComponents.toArray(new String[0]);
+    }
+}
```
I may be mistaken, but i think the [split()](https://www.geeksforgeeks.org/split-string-java-examples/) method does this.

euzintan commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,15 @@
+/**
+ * Custom Exception for the project: duke.Duke
+ */
+public class DukeException extends Exception {
+    public DukeException() {
+        super();
+    }
+
+
+    public DukeException(String message) {
+        super(message);
+    }
+}
```
I've noticed this is your only Exception class. I feel it would be more apt to have multiple Exception classes that extend from your DukeException, each with their own message. For example, InsufficientArgumentsException extends DukeException.

euzintan commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,124 @@
+            default:
+                throw new DukeException("Error: Saved File is badly corrupted");
+            }
+        } catch (Exception err) {
+            throw new DukeException("Error: Saved File is badly corrupted");
+        }
```
Im confused about this part. You are throwing it, to catch it to throw it again without change? unless the constructors throw a different kind of Exception?

Nahoyhp committed:
Assert Check.

Nahoyhp committed:
Add more Assertions cases in following class:

- DukeException
- Storage
- TaskList

Nahoyhp committed:
Merge pull request #1 from Nahoyhp/branch-A-Assertions

Branch A Assertion merge completed

Nahoyhp committed:
Improve Code Quality.

- Use Extract Method to move the logic that handle various commands to new methods.
- Remove nested if-else statements.

Nahoyhp committed:
Merge pull request #2 from Nahoyhp/branch-A-CodeQuality

Improve Code Quality.

Nahoyhp committed:
Add a new command to archive

- Handle as saving current data to specific filePath.

Nahoyhp committed:
Add temp file using for testing into gitIgnore

Nahoyhp committed:
Implement the Load function to read archive files.

Nahoyhp committed:
Add the buffer to close the window in 3 seconds after typing in the word "bye"

Nahoyhp committed:
Fixing of minor bugs.

- Warning when the application is run, due to mismatch in java between javafx and runtime version.
- Unable to load archived file.

Nahoyhp committed:
Add save function.

Nahoyhp committed:
Customize ReadMe.md and CONTRIBUTORS.md file.

Nahoyhp committed:
Add detail for 'bye' in README.md

Nahoyhp committed:
Set theme jekyll-theme-minimal

Nahoyhp committed:
Set theme jekyll-theme-cayman

Nahoyhp committed:
Move the README.md into the folder docs.

Nahoyhp committed:
Merge branch 'master' of https://github.com/Nahoyhp/ip

- Added theme

Nahoyhp committed:
Try to make modification for better styling.

Nahoyhp committed:
Add line break at the end of each table element

Nahoyhp committed:
Changing format for ReadMe.md

Nahoyhp committed:
ReadME.md Part 3

Nahoyhp committed:
README.md Part 4

Nahoyhp committed:
README.md Part 5

Nahoyhp committed:
Fix problem of Dialog Box Shrinking when the screen becomes full.

Nahoyhp committed:
Improve GUI of the application.

- Customize Red box and Image for Error.
- Valid input will reply with green box.

Nahoyhp committed:
Changes avatar images.

Nahoyhp committed:
Refactoring to add in Command Interface

Nahoyhp committed:
Add welcome.png

Nahoyhp committed:
Adding Photo to ReadMe.md Part 1

Nahoyhp committed:
Final Touch-Up

Nahoyhp committed:
Final Touch Up 2

Nahoyhp committed:
Javadoc for Commands.

Nahoyhp committed:
Final Touch-up 3
